pkgs/build-support/docker/default.nix

```
@@ -764,18 +768,24 @@ rec {
         name = "${baseName}-customisation-layer";
         paths = contentsList;
         inherit extraCommands;
+        inherit fakeRootCommands;
+        nativeBuildInputs = [ fakeroot ];
         postBuild = ''
           mv $out old_out
           (cd old_out; eval "$extraCommands" )
 
           mkdir $out
 
-          tar \
-            --sort name \
-            --owner 0 --group 0 --mtime "@$SOURCE_DATE_EPOCH" \
-            --hard-dereference \
-            -C old_out \
-            -cf $out/layer.tar .
+          fakeroot bash -c '
+            set -e
```

```suggestion
          export SHELLOPTS
          fakeroot bash -c '
```

This should make it inherit all shell options.

---

I wonder if it's feasible to `source $stdenv/setup` instead.

---

That seems better as it makes many utility functions available to fakeRootCommands. I changed that line to `source $stdenv/setup`.